lib/WUI/nhttp/file_info.cpp

```diff
@@ -33,6 +35,63 @@ namespace nhttp::printer {
 using namespace handler;
 
 JsonResult FileInfo::DirRenderer::renderState(size_t resume_point, JsonOutput &output, FileInfo::DirState &state) const {
+    if (api == APIVersion::Octoprint)
+        return renderStateOctoprint(resume_point, output, state);
+    else
+        return renderStateV1(resume_point, output, state);
+}
+
+JsonResult FileInfo::DirRenderer::renderStateV1(size_t resume_point, JsonOutput &output, FileInfo::DirState &state) const {
+    struct stat st;
+
+    // Keep the indentation of the JSON in here!
+    // clang-format off
+    JSON_START;
+    JSON_OBJ_START;
+
+        JSON_CONTROL("\"type\":\"FOLDER\"") JSON_COMMA;
+        JSON_FIELD_BOOL("ro", false) JSON_COMMA;
+        // Note: we can only get a timestamp for folders that are not
+        // /usb/ because that is the root of FATFS and stat() don't work on it
+        if (stat(state.filepath, &st) == 0) {
+            JSON_FIELD_INT("m_timestamp", st.st_mtime) JSON_COMMA;
+        }
+        JSON_FIELD_STR("name", state.filename) JSON_COMMA;
+        JSON_FIELD_ARR("children");
+        while (state.dir.get() && (state.ent = readdir(state.dir.get()))) {
+            if (state.ent->d_type == DT_REG and !filename_is_gcode(state.ent->d_name)) {
+                continue;
```

Comment:
It probably isn't part of this task, strictly speaking, but eventually, we want to be showing all the non-Gcode files too. Is the only thing missing for that removing of this `continue`? 😇

Reply:
:thinking:  Kind of yes, we won't send the firmware metadata, but other than that, it is fine. But I asked and Link is not prepared for it as of now, so we should probably leave it here for now and remove it once Link can actually deal with other files.